### DISCUSSION TITLE
Hardware version and compatibility check

### DISCUSSION THREAD

**Discussion by niclask** [rauc/rauc#1338] [General]
I have a use-case that looks like:
An embedded system running Linux that is connected, with a UART, to a MCU. During the update both Linux and MCU have to be updated. Right now there is only one hardware version. In the future there might be a few more hardware versions depending on when the MCU hits end of life. Then the MCU will be replaced with another one which will require a new HEX file. The hardware version will be changed. The update will now contain two HEX files to support both MCUs. 

- The new bundle, which contain both HEX files, is OK to install on a board of hardware version 1.0 and 1.1.
- The older bundles which only contain one HEX file shall only be able to install on a board of the hardware version 1.0.

What is the recommended way to handle the compatibility check? 

Thanks!

**Comment by ejoerns** [comment]
@niclask I'd like to give some generic pointers first. One is the documentation chapter about [Updating Sub-Devices](https://rauc.readthedocs.io/en/latest/advanced.html#updating-sub-devices), the other one is the chapter about [Handling Board Variants](https://rauc.readthedocs.io/en/latest/advanced.html#handling-board-variants-with-a-single-bundle).

If deciding to have the MCU firmware as part of the rootfs, then the compatibility should be checked and resolved there.

In case you decide that variant support would be the way to go, you can simply put an image for the variant to support into the bundle. RAUC will reject the update if it does not have an image for the board's current variant.

Does that already solve your question?

**Reply by niclask** [reply]
Thanks for the reply!

Yes, it does answer my question. I think that storing the HEX-files in the rootfs to hansle rollback sounds like a good solution. To handle the compatibility check, I could make two board variants that points to the same rootfs file.
